### ISSUE TITLE
Tags are not registered if the app is sent to the background within seconds after sendTag(s) is called

### ISSUE THREAD

**Issue by cnordvik** [OneSignal/OneSignal-iOS-SDK#502]
SDK version: 2.10.0

If a user enters the app, enables a few alerts and exits the app by pressing home button then the tags are not registered until the user opens the app at a later stage. How are we supposed to handle this scenario? Is it possible to offer a "send immediately" flag in the SDK in cases like this? It seems you delay sending of the tags to avoid extra http requests but in cases like this it isn't an ideal solution. 

Only workaround I've found is to send them to our server and make our server to a server->server registering of the tags. But this seems like something the SDK should handle? Or am I missing something here? 

```
 OneSignal.sendTags(tags, onSuccess: { [weak self] (_) in
                self?.isLoading = false
            }) { [weak self] (error) in
                self?.isLoading = false
            }
```

**Comment by Nightsd01** [comment]
@cnordvik unfortunately some developers abuse tags and call `sendTags` with an extremely high frequency. To avoid hammering OneSignal servers with unnecessary requests, the SDK batches tag updates together and sends every few seconds. 

In general this is a relatively uncommon situation. If your use case requires tags to be sent _immediately_ I would question how you are using tags, should they really be used to hold critical time sensitive information? Regardless, in this case, I would recommend creating an endpoint on your server to relay the requests as you suggested.

**Comment by cnordvik** [comment]
This is a livescore app, so if a user enters a match and wants goal alerts he turns them on, exits the app and expects to receive the alerts. 

I would imagine this to be a common scenario?
- If a user enables news alerts in a news app for a particular topic and exits the app
- Onboarding of a time senstive event. Do you want alerts for todays Champions League final? User clicks Yes, exits app. Later that day, no events. 

We've used a lot of SDKs like Urban Airship, Azure Notification Hubs ++ and none of them have this behavior. 

I understand why you want to stop the apps from hammering your server, but it seems like a more sensible idea to just queue up the requests on your side and batch them together at the serverside. 

But I guess this isn't changing then? So then we will see if we can rewrite this to do it on our backend instead which is a bit sad since we recently rewrote everything away from the backend to the client since it was the recommended approach :-/

**Comment by rgomezp** [comment]
@cnordvik ,
We've made an internal note to update the SDK to allow tag synchronization to finish even if the app is backgrounded. Thanks for surfacing this issue.

**Comment by cnordvik** [comment]
Thanks for that! Any chance to either keep this issue open and close it when implemented or give me a hint on any planned release-dates for this feature? We're getting so many complaints that we have to do it server side ourselves if the date for a fix is far off...

**Comment by rgomezp** [comment]
@cnordvik , we can't give you a release-date right now unfortunately. Apologies in advance

**Comment by cnordvik** [comment]
@rgomezp Any idea if this is just pushed into "Icebox" or will be part of some release? We get complaints every week about people turning off alerts and still getting them since they turn off alerts for a match they're watching and then exit the app immediately. This causes them to still receive goal alerts which is very annoying for the users. If it is still not any plans to fix it we'll do it on our end but wanted to check in case a fix wasn't far off.

**Comment by rgomezp** [comment]
@cnordvik ,
Unfortunately this isn't something we plan to fix very soon

**Comment by rgomezp** [comment]
Will close this issue for now. We will continue tracking this issue internally. Thanks for your understanding

**Comment by cnordvik** [comment]
Hi, 

Still an ongoing issue for us. Do we just need to switch to the server-API? I guess it will cause more load on your servers if we do that as we won't have any fancy caching strategy but if this issue isn't planned to fix anytime soon I think we have to do that. We used Airship before and they didn't have this issue :-/

**Comment by emawby** [comment]
Hello this is fixed as of release 3.4.1
